app/src/main/kotlin/com/flxrs/dankchat/data/NotificationService.kt

```diff
@@ -254,10 +263,22 @@ class NotificationService : Service(), CoroutineScope {
             ttsMessageQueue -> TextToSpeech.QUEUE_ADD
             else            -> TextToSpeech.QUEUE_FLUSH
         }
-        val ttsMessage = when (name) {
+        var ttsMessage = when (name) {
             previousTTSUser -> message
             else            -> messageFormat.also { previousTTSUser = name }
         }
+        if (removeURL) {
+            val urlPattern = Pattern.compile("((https?|ftp|gopher|telnet|file|Unsure|http):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?\\+-=\\\\\\.&]*)", Pattern.CASE_INSENSITIVE)
+            ttsMessage = ttsMessage.replace(urlPattern.toRegex(), "").trim()
+        }
+        if (removeEmote) {
+            for (emote in this.emotes) {
+                if (ttsMessage.contains(emote.code, ignoreCase = true)) {
```

Comment:
Can't we just directly call replace and skip the `contains` check here?

Reply:
Fixed in [ec9b3b4](https://github.com/flex3r/DankChat/pull/309/commits/ec9b3b43d76619f365ac3eefdff7524b85b354c9)